apps/rebar/src/rebar_resource_v2.erl

```diff
@@ -133,16 +133,11 @@ format_error({no_resource, Source}) ->
 is_resource_type(Type, Resources) ->
     lists:any(fun(#resource{type=T}) -> T =:= Type end, Resources).
 
--spec get_resource_type(term(), [resource()]) -> {ok, resource()}.
-get_resource_type({Type, Location}, Resources) ->
-    get_resource(Type, Location, Resources);
-get_resource_type({Type, Location, _}, Resources) ->
-    get_resource(Type, Location, Resources);
-get_resource_type({Type, _, _, Location}, Resources) ->
-    get_resource(Type, Location, Resources);
-get_resource_type(Location={Type, _, _, _, _, _}, Resources) ->
-    get_resource(Type, Location, Resources);
-get_resource_type(Source, _) ->
+-spec get_resource_type(tuple(), [resource()]) -> {ok, resource()}.
+get_resource_type(Source, Resources) when tuple_size(Source) > 0 ->
+    Type = element(1, Source),
+    get_resource(Type, Source, Resources);
```

Comment:
Location and Source aren't the same. In some cases the Location is pulled out of the tuple.

Reply:
Thanks for the feedback! The reason I originally flattened everything is that a custom resource of size 2-4 can have whatever it likes in its tuple elements, so I thought that being picky about Source and Location is maybe too specific, especially if all this generates is an error message (which may not be the case).

I restored the existing clauses, and merely extended the size 6 case to cover all sizes > 4. The main motivation for the PR is to properly add support a custom resource I am using for some private code, which needs 5 args and previously raised a mere `Failed to fetch and copy dep:`.